### PR TITLE
QCQMC Part 8: Add overlap analysis code

### PR DESCRIPTION
**Pull request by fdmalone**
Final part which provides functionality to compute wavefunction / energy from overlaps constructed via shadow tomography

### CODE REVIEW

File: recirq/qcqmc/analysis.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    def path_string(self) -> str:
+        return self.path_prefix + config.OUTDIRS.DEFAULT_ANALYSIS_DIRECTORY + self.name
+
+    def _json_dict_(self):
+        return attrs.asdict(self)
+
+
+@attrs.frozen(eq=False)
+class OverlapAnalysisData(data.Data):
+    """Container for analysis of overlap.
```
Comment (dstrain115) [inline recirq/qcqmc/analysis.py:71]: This is a pretty short doc string for this class.  It basically just restates the name.

File: recirq/qcqmc/analysis_test.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    )
+
+    print(energy_1)
+    print(energy_2)
+    print(trial_wf.ansatz_energy)
+    print(trial_wf.ansatz_energy - energy_1)
+    print(trial_wf.ansatz_energy - energy_2)
+
+    print(analysis_1.reconstructed_wf_for_k["1"])
+    print(analysis_2.reconstructed_wf_for_k["1"])
```
Comment (dstrain115) [inline recirq/qcqmc/analysis_test.py:137]: Remove print statements?

**Review by dstrain115 — APPROVED** [review]
nits then LGTM.

File: recirq/qcqmc/analysis.py (hunk 1 above)
Comment (fdmalone) [reply recirq/qcqmc/analysis.py]: Fixed.

File: recirq/qcqmc/analysis_test.py (hunk 2 above)
Comment (fdmalone) [reply recirq/qcqmc/analysis_test.py]: Done.

**Pull request merged**